src/App.tsx

```diff
@@ -15,71 +16,103 @@ export const goods = [
+      ? this.setState({ selectedGood: '' })
+      : this.setState({ selectedGood: good })
+  );
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render() {
+    const { selectedGood } = this.state;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        {
+          selectedGood
+            ? (
+              <h1 className="title is-flex is-align-items-center">
+                {`${selectedGood} is selected`}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+                <button
+                  data-cy="ClearButton"
+                  type="button"
+                  className="delete ml-3"
+                  onClick={this.handleClearGood}
+                />
+              </h1>
+            )
+            : (
+              <h1 className="title">No goods selected</h1>
+            )
+        }
+        <table className="table">
+          <tbody>
+            {
+              goods.map(good => {
+                const isSelected = selectedGood === good;
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+                return (
+                  <tr
+                    data-cy="Good"
+                    key={good}
+                    className={classNames(
+                      '',
+                      {
+                        'has-background-success-light':
+                          isSelected,
+                      },
+                    )}
+                  >
+                    <td>
+                      <button
+                        data-cy={
+                          isSelected
+                            ? 'RemoveButton'
+                            : 'AddButton'
+                        }
+                        type="button"
+                        className={classNames(
+                          'button',
+                          {
+                            'is-blue is-info': isSelected,
+                          },
+                        )}
+                        onClick={() => {
+                          this.handleSelectGood(good);
+                        }}
+                      >
+                        {
+                          isSelected
+                            ? '-'
+                            : '+'
+                        }
+                      </button>
```

Comment:
It looks complex with too much logic as for me, everything is good, this way we don't repeat ourselves, but we have only 2 variations. it's not necessary to put condition into each attribute